modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/pagemem/PagesWriteThrottleSandboxTest.java

```diff
@@ -172,14 +194,27 @@ public void testThrottle() throws Exception {
                 }
             }, "metrics-view");
 
+            final boolean sawForm = false;
+
             try (IgniteDataStreamer<Object, Object> ds = ig.dataStreamer(CACHE_NAME)) {
                 ds.allowOverwrite(true);
 
-                for (int i = 0; i < keyCnt * 10; i++) {
-                    ds.addData(ThreadLocalRandom.current().nextInt(keyCnt), new TestValue(ThreadLocalRandom.current().nextInt(),
-                        ThreadLocalRandom.current().nextInt()));
+                while (true) {
+                    long tensOfSecondsPassed = TimeUnit.NANOSECONDS.toSeconds(System.nanoTime() - startNanos) / 10;
+                    if (sawForm && tensOfSecondsPassed % 2 == 1) {
+                        System.out.println("... sleeping ...");
+                        Thread.sleep(1000);
+                    }
+                    else {
+                        ds.addData(ThreadLocalRandom.current().nextInt(keyCnt), new TestValue(ThreadLocalRandom.current().nextInt(),
+                            ThreadLocalRandom.current().nextInt()));
 
-                    putRate.increment();
+                        putRate.increment();
+                        putCount.incrementAndGet();
+                    }
+
+                    if (System.nanoTime() - startNanos > TimeUnit.MINUTES.toNanos(10))
```

Comment:
This test is only run manually when you want to see with your eyes how throttling happens. The test duration is 10 minutes. The duration is chosen as a sufficient, but not too long duration, and it's good to have it fixed because it becomes easier to compare the results of different runs.